### PR TITLE
Replace abi.encodeWithSelector & abi.encodeWithSignature with abi.encodeCall

### LINKED ISSUE #3693: Replace `abi.encodeWithSelector` by `abi.encodeCall`
**🧐 Motivation**
Since 0.8.11, `abi.encodeCall` provide type-safe encode utility comparing with `abi.encodeWithSelector`.

**📝 Details**
Example below and detail are from [DevTools Summit | Topics in Solidity - Harikrishnan Mulackal](https://www.youtube.com/watch?v=CDkJ8rIfqWg).

`abi.encodeWithSelector` can use with `interface.<function>.selector` to prevent typo error, but it doesn't provide type checking.

```solidity
interface miniERC20 {
  function transfer(address to, uint256 value) external;
}

// works successfully
function transferData(uint256 to, uint256 value)
    public
    pure
    returns (bytes memory)
{
    return abi.encodeWithSelector(miniERC20.transfer.selector, to, value);
}
```

`abi.encodeCall` provide type checking during compile time.

```solidity
function transferData(uint256 to, uint256 value)
    public
    pure
    returns (bytes memory)
{
    return abi.encodeCall(miniERC20.transfer, (to, value));
}
```

result
```
error[5407]: TypeError: Cannot implicitly convert component at position 0 from "uint256" to "address".
```

### PR DESCRIPTION
**Pull request by balajipachai**
* adds changeset
* updates contracts: MultiCallTest, ReentrancyMock, UUPSLegacy, ERC4626, SafeERC20, SignatureChecker, ERC165Checker
* updates utils ShortStrings.t.sol

Fixes #3693  

- All occurrences of abi.encodeWithSelector & abi.encodeWithSignature are replaced by abi.encodeCall

#### PR Checklist
1. All tests are passing
2. npm run slither returned `INFO:Slither:. analyzed (429 contracts with 46 detectors), 0 result(s) found`
3. No linting issues
4. All contracts are compiling

- [All Passing] Tests
- [No Changes] Documentation
- [Added] Changeset entry (run `npx changeset add`)

### CODE REVIEW

File: .changeset/big-plums-cover.md
Comment (Amxx) [inline .changeset/big-plums-cover.md:1]: Changeset needs to be one liner to be included in the changelog. If further details are needed then additional section can be added directly in `CHANGELOG.md` ... but that is not the case for this PR IMO

Comment (balajipachai) [reply .changeset/big-plums-cover.md:1]: Done

**Review by Amxx — APPROVED** [review]
LGTM

File: test/utils/ShortStrings.t.sol [hunk 1]
```diff
@@ -25,7 +25,7 @@ contract ShortStringsTest is Test {
 
     function testRevertLong(string memory input) external {
         vm.assume(!_isShort(input));
-        vm.expectRevert(abi.encodeWithSelector(ShortStrings.StringTooLong.selector, input));
+        vm.expectRevert(abi.encodeCall(ShortStrings.StringTooLong, (input)));
```
Comment (Amxx) [inline test/utils/ShortStrings.t.sol:28]:
```suggestion
        vm.expectRevert(abi.encodeWithSelector(ShortStrings.StringTooLong.selector, input));
```

File: test/utils/ShortStrings.t.sol [hunk 2]
```diff
@@ -25,7 +25,7 @@ contract ShortStringsTest is Test {
 
     function testRevertLong(string memory input) external {
         vm.assume(!_isShort(input));
-        vm.expectRevert(abi.encodeWithSelector(ShortStrings.StringTooLong.selector, input));
```
[the remark this answers is not in the document]
Comment (Amxx) [reply test/utils/ShortStrings.t.sol:28]: https://github.com/ethereum/solidity/issues/14287

File: test/utils/ShortStrings.t.sol (hunk 1 above)
Comment (balajipachai) [reply test/utils/ShortStrings.t.sol:28]: Thank You for pointing this out @Amxx, however, when I ran `npm run test` all tests were passing.
I don't see anything related to running foundry tests, thus, missed this one.
Committing this suggestion and resolving it.
Thanks again.

**Review by Amxx — APPROVED** [review]
LGTM

**Review by frangio — APPROVED** [review]
LGTM. Thanks!

**Pull request merged**